Build failure: ghostscript-10.04.0

### Steps To Reproduce

Steps to reproduce the behavior:
1. build ghostscript-10.04.0

### Build log

```
rewriting symlink /nix/store/58s4hjjydcvld4zcqha58ziwjybkfyvw-ghostscript-10.04.0-man/share/man/man1/gsbj.1.gz to be relative to /nix/store/58s4hjjydcvld4zcqha58ziwjybkfyvw-ghostscript-10.04.0-man
rewriting symlink /nix/store/58s4hjjydcvld4zcqha58ziwjybkfyvw-ghostscript-10.04.0-man/share/man/man1/gsdj.1.gz to be relative to /nix/store/58s4hjjydcvld4zcqha58ziwjybkfyvw-ghostscript-10.04.0-man
rewriting symlink /nix/store/58s4hjjydcvld4zcqha58ziwjybkfyvw-ghostscript-10.04.0-man/share/man/man1/eps2eps.1.gz to be relative to /nix/store/58s4hjjydcvld4zcqha58ziwjybkfyvw-ghostscript-10.04.0-man
rewriting symlink /nix/store/58s4hjjydcvld4zcqha58ziwjybkfyvw-ghostscript-10.04.0-man/share/man/man1/gsdj500.1.gz to be relative to /nix/store/58s4hjjydcvld4zcqha58ziwjybkfyvw-ghostscript-10.04.0-man
rewriting symlink /nix/store/58s4hjjydcvld4zcqha58ziwjybkfyvw-ghostscript-10.04.0-man/share/man/man1/gslj.1.gz to be relative to /nix/store/58s4hjjydcvld4zcqha58ziwjybkfyvw-ghostscript-10.04.0-man
rewriting symlink /nix/store/58s4hjjydcvld4zcqha58ziwjybkfyvw-ghostscript-10.04.0-man/share/man/man1/ps2pdf14.1.gz to be relative to /nix/store/58s4hjjydcvld4zcqha58ziwjybkfyvw-ghostscript-10.04.0-man
rewriting symlink /nix/store/58s4hjjydcvld4zcqha58ziwjybkfyvw-ghostscript-10.04.0-man/share/man/man1/ps2pdf13.1.gz to be relative to /nix/store/58s4hjjydcvld4zcqha58ziwjybkfyvw-ghostscript-10.04.0-man
rewriting symlink /nix/store/58s4hjjydcvld4zcqha58ziwjybkfyvw-ghostscript-10.04.0-man/share/man/man1/ps2pdf12.1.gz to be relative to /nix/store/58s4hjjydcvld4zcqha58ziwjybkfyvw-ghostscript-10.04.0-man
@nix { "action": "setPhase", "phase": "installCheckPhase" }
Running phase: installCheckPhase
dyld[68823]: Library not loaded: libgs.10.dylib
  Referenced from: <497CFDBA-A6D1-332C-B983-C3CFF6293055> /nix/store/ag7vqbvj1pk7lqczyc3h9ynwv3y199z8-ghostscript-10.04.0/bin/gs
  Reason: tried: 'libgs.10.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OSlibgs.10.dylib' (no such file), 'libgs.10.dylib' (no such file), '/usr/local/lib/libgs.10.dylib' (no such file), '/usr/lib/libgs.10.dylib' (no such file, not in dyld cache), '/private/tmp/nix-build-ghostscript-10.04.0.drv-0/ghostscript-10.04.0/libgs.10.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/private/tmp/nix-build-ghostscript-10.04.0.drv-0/ghostscript-10.04.0/libgs.10.dylib' (>
/nix/store/vrjr8rg1gsvvncgdl7qybh5h7vgqdxvn-stdenv-darwin/setup: line 1596: 68823 Abort trap: 6           $out/bin/gs --version
```

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@tobim

### Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->
```
 - system: `"aarch64-darwin"`
 - host os: `Darwin 24.1.0, macOS 15.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.18.8`
 - nixpkgs: `/nix/store/dxdcvjnvz3b91gvsrhpb7gp156nnj8bf-source`

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

Issue appears linked to any application that requires python e.g. ansible (as mentioned by @ironicbadger), borgmatic, kitty etc.

My build work fine when I remove any of those apps.

I have tried applying the overlay and the build no longer fails on ghostscript-10.04.0 but then fails on bulding some apps based on python e.g.

```
Building nix-darwin config...
nix --extra-experimental-features 'nix-command flakes'  build ".#darwinConfigurations.Virtuous-Dynamite.system" 
error: builder for '/nix/store/zbal30xk74avsmlgavbbw06zvklxs8sd-borgbackup-1.4.0.drv' failed with exit code 1;
       last 10 log lines:
       > /nix/store/xklks5naxqwf3vbnnf4slppymnwl1shi-borgbackup-1.4.0/lib/python3.11/site-packages/borg/testsuite/xattr.py:64:
       > _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
       > /nix/store/xklks5naxqwf3vbnnf4slppymnwl1shi-borgbackup-1.4.0/lib/python3.11/site-packages/borg/testsuite/xattr.py:28: in assert_equal_se
       >     self.assert_equal(is_x, want_x)
       > E   AssertionError: Items in the first set but not the second:
       > E   b'com.apple.provenance'
       > =========================== short test summary info ============================
       > FAILED xattr.py::XattrTestCase::test - AssertionError: Items in the first set but not the second:
       > FAILED xattr.py::XattrTestCase::test_listxattr_buffer_growth - AssertionError: Items in the first set but not the second:
       > ============ 2 failed, 1290 passed, 278 skipped in 91.88s (0:01:31) ============
       For full logs, run 'nix log /nix/store/zbal30xk74avsmlgavbbw06zvklxs8sd-borgbackup-1.4.0.drv'.
error: 1 dependencies of derivation '/nix/store/9p5gfwki4035v6g3grj8ry08rsljj7gr-system-applications.drv' failed to build
error: 1 dependencies of derivation '/nix/store/ahcs5rkj4n30cg4p3rk7y1yfqlr9gnb6-darwin-system-24.05.20241116.e8c38b7+darwin4.698a62c.drv' failed to build
error: Recipe `build` failed on line 11 with exit code 1
```

I excluded borgbackup and tried again, ansible installed ok but it took a while. It appeared to build the package from source.

--- a/pkgs/misc/ghostscript/default.nix
+++ b/pkgs/misc/ghostscript/default.nix
@@ -104,6 +104,8 @@ stdenv.mkDerivation rec {
     sed "s@^ZLIBDIR=.*@ZLIBDIR=${zlib.dev}/include@" -i configure.ac
 
     autoconf
+  '' + lib.optionalString stdenv.hostPlatform.isDarwin ''
+    export DARWIN_LDFLAGS_SO_PREFIX=$out/lib/
   '';
 
   configureFlags = [
@@ -140,19 +142,8 @@ stdenv.mkDerivation rec {
     done
   '';
 
-  # dynamic library name only contains maj.min, eg. '9.53'
-  dylib_version = lib.versions.majorMinor version;
-  preFixup = lib.optionalString stdenv.isDarwin ''
-    install_name_tool -change libgs.dylib.$dylib_version $out/lib/libgs.dylib.$dylib_version $out/bin/gs
-    install_name_tool -change libgs.dylib.$dylib_version $out/lib/libgs.dylib.$dylib_version $out/bin/gsx
-  '';
-
   # validate dynamic linkage
   doInstallCheck = true;
-  preInstallCheck = if stdenv.hostPlatform.isDarwin then ''
-    DYLD_LIBRARY_PATH=$out/lib
-    export DYLD_LIBRARY_PATH
-  '' else null;
   installCheckPhase = ''
     runHook preInstallCheck
 

